### PR TITLE
💬 Add section for team management topics

### PR DESCRIPTION
**Pull request by olyism**
Add navigation menu for product-agnostic team management pages, so we can group pages with similar purpose and support more team management feature Docs pages to come.

Summary of changes:
- Add 'Team management' parent nav menu to the left navigation menu
  - Add simple Overview page for navigation and link to 'Manage API token' over under the APIs section
  - Move 2FA and User Permissions to this new parent nav menu, plus redirects
- Add 'Manage API token' on the homepage reference section

Note: this is an interim solution to have a place for product-agnostic team management Docs pages, while we're still working on improving the overall navigation.

Screenshots:

Screenshot 2023-09-19 at 4 38 52 PM

Screenshot 2023-09-19 at 4 31 34 PM

### CODE REVIEW

**Comment by olyism** [comment]
hi @jameshill here's the Docs PR to create an interim nav menu for team management related features.

For future team management related features you can put the markdown file under `pages/team_management` and add the navigation under [Team management](https://github.com/buildkite/docs/pull/2483/files#diff-563edf1b408959f1be664c0e40280fed93adc3959e4cdb2254e900a27b71cd04R309) in `nav.yml`.

Because _Manage API token_ is under the APIs section, I've added this as a link in the [Team Management Overview page](https://github.com/buildkite/docs/pull/2483/files#diff-011c97c70c61b939028d87c338b9608583a0bba31312e2b1cdb33372fceaa95eR11).

File: config/routes.rb [hunk 1]
```diff
@@ -73,6 +74,7 @@
   get "/docs/tutorials/bitbucket",                to: redirect("/docs/integrations/bitbucket")
   get "/docs/tutorials/custom-saml",              to: redirect("/docs/integrations/sso/custom-saml")
   get "/docs/tutorials/sso-setup-with-graphql",   to: redirect("/docs/integrations/sso/sso-setup-with-graphql")
+  get "/docs/tutorials/2fa",                      to: redirect("/docs/team-management/2fa")
```
Comment (dannymidnight) [inline config/routes.rb:77]: In addition to adding the redirect I also suggest updating existing URLs to use the new path.

Comment (dannymidnight) [reply config/routes.rb:77]: As an aside (unrelated to this PR), I noted redirects are ignored on the [preview site](https://2483--bk-docs-preview.netlify.app/docs/tutorials/2fa), which makes sense.

Comment (jameshill) [reply config/routes.rb:77]: I reckon we could leave this 2FA tutorial out of the team-management, the benefit of leaving it under tutorials is that we can add the "enforce 2fa" documentation under team-management without it being confusing for readers having them side by side.

Comment (olyism) [reply config/routes.rb:77]: > In addition to adding the redirect I also suggest updating existing URLs to use the new path.
Thanks @dannymidnight, I missed existing links, good catch and I'll push up a fix.

File: app/views/homepage/_references.html.erb [hunk 2]
```diff
@@ -57,7 +57,11 @@
         {
           title: "Webhooks",
           link: "/docs/apis/webhooks"
-        }
+        },
+        {
+          title: "Managing API access tokens",
+          link: "/docs/apis/managing-api-tokens"
```
Comment (jameshill) [inline app/views/homepage/_references.html.erb:63]: to keep this PR tight I reckon you could remove this from the footer.

**Comment by olyism** [comment]
> I reckon we could leave this 2FA tutorial out of the team-management, the benefit of leaving it under tutorials is that we can add the "enforce 2fa" documentation under team-management without it being confusing for readers having them side by side.

hi @jameshill I reverted 2FA back under the tutorials menu. However, we only have one page under the `team-management` menu. If released, it may give an impression that we have just one feature for team management.

Screenshot 2023-09-19 at 5 27 05 PM

That being said, if you're happy with the navigation structure then I can put this PR back into draft, and you can use the changes in this branch to add the new pages.

**Comment by olyism** [comment]
> Hey @olyism i've got two fast follow PRs to add additional features to that section.

Sounds good @jameshill 👍  let's quickly get this merged to unblock your follow up PRs.

File: pages/team_management.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+toc: false
+---
+
+# Team management overview
```
Comment (MelissaKaulfuss) [inline pages/team_management.md:5]:
```suggestion
# Team management
```

File: pages/team_management.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+Team management is fundamental in CI/CD tools to ensure streamlined processes, proper access controls, and efficient collaboration. Buildkite provides features for your team management needs:
```
Comment (MelissaKaulfuss) [inline pages/team_management.md:7]:
```suggestion
Managing users and teams in CI/CD is fundamental to collaboration, streamlined processes, and ensuring adequate access controls. Buildkite provides features to manage team access:
```

**Review by MelissaKaulfuss — COMMENTED** [review]
Just some slight tweaks to the words in Team Management.
Looks great!

**Comment by olyism** [comment]
Thanks @MelissaKaulfuss, great wording improvements, I've committed your suggestions 🙌

**Review by MelissaKaulfuss — APPROVED** [review]
Looking good @olyism 👍🏻

**Pull request merged**